src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -136,27 +130,42 @@ export function findNodeKeyUp(editorView: EditorView, pos: number): NodeKey {
  * ProseMirror will render content nodes into this element.
  */
 export function createReactNodeViewConstructor(
-  reactNodeViewConstructor: ReactNodeViewConstructor,
+  nodeViewConstructor: ReactNodeViewConstructor,
   registerPortal: RegisterPortal
 ) {
-  function nodeViewConstructor(
+  function nodeViewConstructorWrapper(
     node: Node,
     editorView: EditorView,
     getPos: () => number,
     decorations: readonly Decoration[],
     innerDecorations: DecorationSource
   ): NodeView {
-    const reactNodeView = reactNodeViewConstructor(
+    const nodeView = nodeViewConstructor(
       node,
       editorView,
       getPos,
       decorations,
       innerDecorations
     );
 
-    let componentRef: NodeViewWrapperRef | null = null;
+    const { component: NodeView } = nodeView;
+    if (!NodeView) {
```

Comment:
worth adding a comment here to note that this is the branch which continues to allow specifying non-React nodeviews?